0-common/schema-ddl/src/main/scala/com.snowplowanalytics/iglu.schemaddl/jsonschema/SanityLinter.scala

```diff
@@ -115,45 +115,55 @@ object SanityLinter {
    * @return non-empty list of summed failures (all, including nested) or
    *         unit in case of success
    */
-  def lint(schema: Schema, severityLevel: SeverityLevel): LintSchema = {
+  def lint(schema: Schema, severityLevel: SeverityLevel, height: Int = 0): LintSchema = {
     // Current level validations
     val validations = severityLevel.linters.map(linter => linter(schema))
       .foldMap(_.toValidationNel)
 
+    val rootTypeCheck = (height match {
+      case 0 =>
+        schema.`type` match {
+          case Some(Object) => propertySuccess
```

Comment:
Can we also add condition that it must have `properties`? Because `object` without `properties` also isn't particularly useful for DDL-generation.